File: plugins/sbt/sbt-codegen-example/src/main/scala/io/github/dreamylost/service/QueryResolverImplMain.java
package io.github.dreamylost.service;

import io.github.dreamylost.model.DroidEntity;
import io.github.dreamylost.model.EpisodeEntity;
import io.github.dreamylost.model.HumanEntity;

import java.util.List;

/**
 * @author liguobin@growingio.com
 * @version 1.0, 2020/7/16
 * @see [[server code  https://github.com/jxnu-liguobin/springboot-examples/tree/master/graphql-complete]]
 */
public class QueryResolverImplMain {

    public static void main(String[] args) throws Exception {
        System.out.println("=======get droid id 2001=========");
        QueryResolverImpl droidResolver = new QueryResolverImpl();
        DroidEntity d = droidResolver.droid("2001");
        System.out.println(d);

        System.out.println("=======get humans all=======");
        List<HumanEntity> hums = droidResolver.humans();
        for (HumanEntity h : hums) {
            System.out.println(h);
        }

        System.out.println("=======get human id 1002=======");
        HumanEntity hum = droidResolver.human("1002");
        System.out.println(hum);


        System.out.println("=======get hero Episode.EMPIRE=======");
        io.github.dreamylost.model.CharacterEntity character = droidResolver.hero(EpisodeEntity.EMPIRE);
        System.out.println(character);


    }

}

File: plugins/sbt/graphql-java-codegen-sbt-plugin/src/sbt-test/graphql-codegen-sbt-plugin/example-client/src/main/scala/io/github/dreamylost/service/QueryResolverImplMain.scala
package io.github.dreamylost.service

import io.github.dreamylost.model.EpisodeDO

import scala.collection.JavaConverters._

/**
 *
 * @author liguobin@growingio.com
 * @version 1.0,2020/7/21
 */
object QueryResolverImplMain extends App {

  val droidResolver = new QueryResolverImpl
  println("=======QueryResolverImpl=======")
  // need typename on Projection
  println("=======get droid id 2001=========")
  val d = droidResolver.droid("2001")
  println(d)

  println("=======get humans all=======")
  val hums = droidResolver.humans
  for (h <- hums.asScala) {
    println(h)
  }

  println("=======get human id 1002=======")
  val hum = droidResolver.human("1002")
  println(hum)

  println("=======get hero Episode.EMPIRE=======")
  val character = droidResolver.hero(EpisodeDO.EMPIRE)
  println(character)

  println("=======QueryResolverImpl2=======")
  /**
   * MAX DEPTH = 4
   * gql:{{{
   * { id name friends {
   *     id name friends {
   *         id name friends {
   *             id name friends {
   *                 id name friends {
   *                      id name appearsIn secretBackstory __typename }
   *                 appearsIn secretBackstory __typename }
   *             appearsIn secretBackstory __typename }
   *         appearsIn secretBackstory __typename }
   *     appearsIn secretBackstory __typename }
   * appearsIn primaryFunction secretBackstory __typename }
   * }}}
   */
  val droidResolver2 = new QueryResolverImpl2
  println("=======2get droid id 2001=========")
  val d2 = droidResolver2.droid("2001")
  println(d2)

  println("=======2get humans all=======")
  val hums2 = droidResolver2.humans
  for (h <- hums2.asScala) {
    println(h)
  }

  println("=======2get human id 1002=======")
  val hum2 = droidResolver2.human("1002")
  println(hum2)

  println("=======2get hero Episode.EMPIRE=======")
  val character2 = droidResolver2.hero(EpisodeDO.EMPIRE)
  println(character2)

}
